app/src/main/java/io/apicurio/registry/rest/v2/AdminResourceImpl.java

```diff
@@ -268,4 +290,64 @@ public void write(OutputStream os) throws IOException, WebApplicationException {
         return Response.ok(stream).type("application/zip").build();
     }
 
+    /**
+     * @see io.apicurio.registry.rest.v2.AdminResource#createRoleMapping(io.apicurio.registry.rest.v2.beans.RoleMapping)
+     */
+    @Override
+    @Authorized(style=AuthorizedStyle.None, level=AuthorizedLevel.Admin)
+    @RoleBasedAccessApiOperation
+    public void createRoleMapping(RoleMapping data) {
+        storage.createRoleMapping(data.getPrincipalId(), data.getRole().name());
+    }
+
+    /**
+     * @see io.apicurio.registry.rest.v2.AdminResource#listRoleMappings()
+     */
+    @Override
+    @Authorized(style=AuthorizedStyle.None, level=AuthorizedLevel.Admin)
+    @RoleBasedAccessApiOperation
+    public List<RoleMapping> listRoleMappings() {
+        List<RoleMappingDto> mappings = storage.getRoleMappings();
+        return mappings.stream().map(dto -> {
+            RoleMapping mapping = new RoleMapping();
+            mapping.setPrincipalId(dto.getPrincipalId());
```

Comment:
Extract to a mapper and use method reference? Possibly not necessary since I can only see two occurrences.

Reply:
Done.